main.go

```diff
@@ -103,15 +128,141 @@ func main() {
 		}
 
 		server := http.Server{TLSConfig: tlsConfig}
-		tlsListener := tls.NewListener(listener, tlsConfig)
-		server.Serve(tlsListener)
-	} else {
-		server := http.Server{}
-		server.Serve(listener)
+		tlsListener := tls.NewListener(httpsListener, tlsConfig)
+
+		go func() {
+			err := server.Serve(tlsListener)
+			if err != nil {
+				abort(err.Error())
+			}
+		}()
 	}
+
+	// Block forever. The serve Goroutines above will abort the program if
+	// either of them fails.
+	select {}
 }
 
-// ---
+//
+// Private types
+//
+
+// options is a container for the command line options passed to stripe-mock.
+type options struct {
+	fixturesPath string
+
+	http           bool
+	httpPort       int
+	httpUnixSocket string
+
+	https           bool
+	httpsPort       int
+	httpsUnixSocket string
+
+	port        int
+	showVersion bool
+	specPath    string
+	unixSocket  string
+}
+
+func (o *options) checkConflictingOptions() error {
+	if o.unixSocket != "" && o.port != 0 {
+		return fmt.Errorf("Please specify only one of -port or -unix")
+	}
+
+	//
+	// HTTP
+	//
+
+	if o.http && (o.httpUnixSocket != "" || o.httpPort != 0) {
+		return fmt.Errorf("Please don't specify -http when using -http-port or -http-unix")
+	}
+
+	if (o.unixSocket != "" || o.port != 0) && (o.httpUnixSocket != "" || o.httpPort != 0) {
+		return fmt.Errorf("Please don't specify -port or -unix when using -http-port or -http-unix")
+	}
+
+	if o.httpUnixSocket != "" && o.httpPort != 0 {
+		return fmt.Errorf("Please specify only one of -http-port or -http-unix")
+	}
+
+	//
+	// HTTPS
+	//
+
+	if o.https && (o.httpsUnixSocket != "" || o.httpsPort != 0) {
+		return fmt.Errorf("Please don't specify -https when using -https-port or -https-unix")
+	}
+
+	if (o.unixSocket != "" || o.port != 0) && (o.httpsUnixSocket != "" || o.httpsPort != 0) {
+		return fmt.Errorf("Please don't specify -port or -unix when using -https-port or -https-unix")
+	}
+
+	if o.httpsUnixSocket != "" && o.httpsPort != 0 {
+		return fmt.Errorf("Please specify only one of -https-port or -https-unix")
+	}
+
+	return nil
+}
+
+// getHTTPListener gets a listener on a port or unix socket depending on the
+// options provided. If HTTPS should not be enabled, it returns nil.
```

Comment:
Should this say "If HTTP should not be enabled" instead?

Reply:
Oops. Yep, fixed now. Thanks!